src/service_discovery.rs

```diff
@@ -116,13 +116,32 @@ impl<Exe: Executor> ServiceDiscovery<Exe> {
             } = convert_lookup_response(&response)?;
             is_authoritative = authoritative;
 
-            // use the TLS connection if available
-            let connection_url = if let Some(u) = &broker_url_tls {
-                u.clone()
-            } else if let Some(u) = &broker_url {
-                u.clone()
-            } else {
-                return Err(ServiceDiscoveryError::NotFound);
+            // Use broker url with the same schema of manager.url
+            let (connection_url, broker_url) = match base_url.scheme() {
+                "pulsar+ssl" => {
+                    if let Some(u) = &broker_url_tls {
+                        (
+                            u.clone(),
+                            format!("{}:{}", u.host_str().unwrap(), u.port().unwrap_or(6651)),
+                        )
+                    } else {
+                        return Err(ServiceDiscoveryError::NotFound);
+                    }
+                }
+                "pulsar" => {
+                    if let Some(u) = &broker_url {
+                        (
+                            u.clone(),
+                            format!("{}:{}", u.host_str().unwrap(), u.port().unwrap_or(6650)),
+                        )
+                    } else {
+                        return Err(ServiceDiscoveryError::NotFound);
+                    }
```

Comment:
nit: Not really a problem, but feel it's better to deduplicate the code here. Consider letting `match` return `(broker_url_maybe_tls, broker_port)`, then do `let connection_url = if let Some(..) = &broker_url_maybe_tls {` after.